chia/types/eligible_coin_spends.py

```diff
@@ -97,12 +222,138 @@ def get_deduplication_info(
                     )
                     # Update this mempool item's coin spends map
                     bundle_coin_spends[coin_id] = BundleCoinSpend(
-                        spend_data.coin_spend, spend_data.eligible_for_dedup, spend_data.additions, spend_cost
+                        coin_spend=spend_data.coin_spend,
+                        eligible_for_dedup=spend_data.eligible_for_dedup,
+                        eligible_for_fast_forward=spend_data.eligible_for_fast_forward,
+                        additions=spend_data.additions,
+                        cost=spend_cost,
                     )
                 duplicate_cost = spend_cost
                 # If we end up including this item, update this entry's cost
-                new_eligible_spends[coin_id] = DedupCoinSpend(current_solution, duplicate_cost)
+                new_dedup_spends[coin_id] = DedupCoinSpend(current_solution, duplicate_cost)
             cost_saving += duplicate_cost
         # Update the eligible coin spends data
-        self.eligible_spends.update(new_eligible_spends)
+        self.deduplication_spends.update(new_dedup_spends)
         return unique_coin_spends, uint64(cost_saving), unique_additions
+
+    async def process_fast_forward_spends(
+        self,
+        *,
+        mempool_item: InternalMempoolItem,
+        get_unspent_lineage_info_for_puzzle_hash: Callable[[bytes32], Awaitable[Optional[UnspentLineageInfo]]],
+        height: uint32,
+        constants: ConsensusConstants,
+    ) -> None:
+        """
+        Provides the caller with an in-place internal mempool item that has a
+        proper state of fast forwarded coin spends and additions starting from
+        the most recent unspent versions of the related singleton spends.
```

Comment:
it sounds like `mempool_item` is an in-out-parameter. It might be good to mention that